### PR TITLE
Feature/df 1788 from master inconsistent phone number format

### PR DESCRIPTION
**Pull request by hamidsoleymani**
Alle telefonnummer, die zu job-add kommen werden unformatiert gespeichert. Die die zu avam gehen werden formatiert und dann zu avam geschickt

### CODE REVIEW

File: app-external-job-ad-import-task/src/main/java/ch/admin/seco/jobs/services/jobadservice/integration/external/importer/config/ExternalJobAdvertisementAssembler.java [hunk 1]
```diff
@@ -237,7 +228,7 @@ private ExternalCompanyDto createCompany(Oste externalJobAdvertisement) {
                 externalJobAdvertisement.getUntPostfach(),
                 externalJobAdvertisement.getUntPostfachPlz(),
                 externalJobAdvertisement.getUntPostfachOrt(),
-                externalJobAdvertisement.getUntTelefon(),
+                sanitizePhoneNumber(externalJobAdvertisement.getUntTelefon(), externalJobAdvertisement),
```
Comment (cjpsycos) [inline app-external-job-ad-import-task/src/main/java/ch/admin/seco/jobs/services/jobadservice/integration/external/importer/config/ExternalJobAdvertisementAssembler.java]: Think about only pass fingerprint if rest is irrelevant

File: app-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/JobAdvertisementToAvamAssembler.java [hunk 2]
```diff
@@ -1,318 +1,309 @@
...
+
+    private void fillContact(TOsteEgov avamJobAdvertisement, Contact contact) {
+        if (contact == null) {
+            return;
+        }
+        avamJobAdvertisement.setKpAnredeCode(AvamCodeResolver.SALUTATIONS.getLeft(contact.getSalutation()));
+        avamJobAdvertisement.setKpVorname(contact.getFirstName());
+        avamJobAdvertisement.setKpName(contact.getLastName());
+        avamJobAdvertisement.setKpTelefonNr(sanitizePhoneNumber(contact.getPhone()));
+        // FIXME: Temparory fix for mulitple email-addresses. to be remove after 01.09.2018 or handled otherwise
```
Comment (cjpsycos) [inline app-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/JobAdvertisementToAvamAssembler.java]: missing in fillPublicContact

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/webform/WebformCreateJobAdvertisementDto.java [hunk 3]
```diff
@@ -93,6 +96,7 @@ public ContactDto getContact() {
     }
 
     public WebformCreateJobAdvertisementDto setContact(ContactDto contact) {
+        contact.setPhone(sanitizePhoneNumber(contact.getPhone(), PhoneNumberUtil.PhoneNumberFormat.E164));
```
Comment (cjpsycos) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/webform/WebformCreateJobAdvertisementDto.java]: This should not be done in the DTO (Relfection based ObjectMapper), but should be done in the ApplicationService-Assembler

File: app-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/JobAdvertisementToAvamAssembler.java (hunk 2 above)
Comment (hamidsoleymani) [reply app-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/JobAdvertisementToAvamAssembler.java]: ok

File: app-external-job-ad-import-task/src/main/java/ch/admin/seco/jobs/services/jobadservice/integration/external/importer/config/ExternalJobAdvertisementAssembler.java (hunk 1 above)
Comment (hamidsoleymani) [reply app-external-job-ad-import-task/src/main/java/ch/admin/seco/jobs/services/jobadservice/integration/external/importer/config/ExternalJobAdvertisementAssembler.java]: ok

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/webform/WebformCreateJobAdvertisementDto.java (hunk 3 above)
Comment (hamidsoleymani) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/webform/WebformCreateJobAdvertisementDto.java]: ok

**Comment by hamidsoleymani** [comment]
> * I don't like that we have three versions of the sanitizePhoneNumber() method.
> * Where do you make sure that the phone numbers go to AVAM formatted but are sanitized when we get them back??

It is already implemented, so I did not have to do it.

**Pull request merged**